hack/golangci-lint.sh

```diff
@@ -3,6 +3,23 @@
 # Need to run linter twice to cover all the build tags code paths
 set -e
 
+# Mac's are "special" and don't support `declare -A`
```

Comment:
Comment is misleading. The `-A` far below has nothing to do with anything done in this code block. A more informative comment might be "Test only remote on macs"

Reply:
I put that there in case somebody wonders why there's a whole separate section just for mac.  IMHO, "why" comments are better than "what".

How about: "Dedicated block for Darwin: It doesn't support the rest of this script, only needs to check 'remote', and its golangci-lint needs specialized arguments."

(I have half-a-mind to make a separate `hack/mac_golangci-lint.sh` but that would mean tinkering with the `Makefile` :confused:)